src/agentscope/utils/mixture_of_agent.py

```diff
@@ -0,0 +1,167 @@
+# -*- coding: utf-8 -*-
+# pylint: disable=C0301
+""" Utils for mixing model's answers in agentscope """
+
+from typing import Union, List, Sequence, Tuple
+import concurrent.futures
+
+from agentscope.message import Msg
+from agentscope.models import (
+    ModelWrapperBase,
+    load_model_by_config_name,
+)
+
+
+# Referenced from the project [MoA](https://github.com/togethercomputer/MoA)
+DEFAULT_AGGREGATOR_PROMPT = """You have been provided with a set of responses from various open-source models to the latest user query. Your task is to synthesize these responses into a single, high-quality response. It is crucial to critically evaluate the information provided in these responses, recognizing that some of it may be biased or incorrect. Your response should not simply replicate the given answers but should offer a refined, accurate, and comprehensive reply to the instruction. Ensure your response is well-structured, coherent, and adheres to the highest standards of accuracy and reliability.
+
+Responses from models:"""  # noqa
+
+
+class MixtureOfAgents:
+    """
+    The MoA model that take multiple models and aggregate their responses,
+    leverages the collective strengths of multiple LLMs to enhance performance.
+    Reference from the project [MoA](https://github.com/togethercomputer/MoA).
+    """
+
+    def __init__(
+        self,
+        main_model: Union[str, ModelWrapperBase],
+        reference_models: List[Union[str, ModelWrapperBase]],
+        rounds: int = 1,
+        aggregator_prompt: str = DEFAULT_AGGREGATOR_PROMPT,
+        show_internal: bool = False,
+    ) -> None:
+        """
+        Args:
+            main_model (`Union[str, ModelWrapperBase]`):
+                The main_model used for aggregation.
```

Comment:
The docstring should be more specific, e.g. `The config name of the model who makes the final decision`